dimod/serialization/format.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2019 D-Wave Systems Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+# ================================================================================================
+from itertools import islice
+
+import numpy as np
+
+import dimod
+
+
+class PassVar(object):
+    def __str__(self):
+        return '..'
+
+
+class SampleSetFormatter(object):
+
+    min_column_width = 2
+    column_space = 2
+
+    max_rows = 40
+    max_columns = 12  # 17
+    header = True
+    footer = True
+    index = True
```

Comment:
Instead of class variables (which in this case are basically global variables), I would model the formatter/pretty printer according to [pprint](https://docs.python.org/3/library/pprint.html) module.

Reply:
For documentation we want to be able to set some formatting options once (in the docstring setup) and have them persist. I don't see how that can be done with pprint?

I was following [pandas'](https://pandas.pydata.org/pandas-docs/stable/options.html#overview) pattern. But we could use [numpy's](https://docs.scipy.org/doc/numpy/reference/generated/numpy.set_printoptions.html) instead if you think it's cleaner.

Reply:
They more-or-less amount to the same thing though.